### ISSUE TITLE
"strict-export-declare-modifiers" and "expect" are inconsistent in TypeScript@next

### ISSUE THREAD

**Issue by zbjornson** [Microsoft/dtslint#184]
```ts
declare class Foo {}
export { Foo as Bar }
```
This triggers `strict-export-declare-modifiers   'declare' keyword is redundant here`. However, removing it triggers `expect    TypeScript@next compile error: A 'declare' modifier is required for a top level declaration in a .d.ts file.`.
